external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala

```diff
@@ -138,10 +139,15 @@ class KafkaTestUtils extends Logging {
 
     if (server != null) {
       server.shutdown()
+      server.awaitShutdown()
       server = null
     }
 
-    brokerConf.logDirs.foreach { f => Utils.deleteRecursively(new File(f)) }
+    // On Windows, `logDirs` is left open even after Kafka server above is completely shut-downed
+    // in some cases. It leads to test failures on Windows if these are not ignored.
+    brokerConf.logDirs.map(new File(_))
+      .filterNot(FileUtils.deleteQuietly)
```

Comment:
> Is the point just that the former method is "quiet" instead of throwing an exception

Yes!

I will try to clean up including your another comment.